iml-api/src/graphql.rs

```diff
@@ -220,96 +229,285 @@ impl QueryRoot {
+                "Filesystem not found or MGS is not mounted",
+                Value::null(),
+            ))?;
 
-            let snapshots: Vec<_> = xs
+        let kwargs: HashMap<String, String> = vec![("message".into(), "Mounting snapshot".into())]
+            .into_iter()
+            .collect();
+
+        let jobs = serde_json::json!([{
+            "class_name": "MountSnapshotJob",
+            "args": {
+                "fsname": fsname,
+                "name": name,
+                "fqdn": active_mgs_host_fqdn,
+            }
+        }]);
+        let command_id: i32 = iml_job_scheduler_rpc::call(
+            &context.rabbit_pool.get().await?,
+            "run_jobs",
+            vec![jobs],
+            Some(kwargs),
+        )
+        .map_err(ImlApiError::ImlJobSchedulerRpcError)
+        .await?;
+
+        get_command(&context.pg_pool, command_id)
+            .await
+            .map_err(|e| e.into())
+    }
+    #[graphql(arguments(
+        fsname(description = "Filesystem snapshot was taken from"),
+        name(description = "Name of the snapshot"),
+    ))]
+    async fn unmount_snapshot(
+        context: &Context,
+        fsname: String,
+        name: String,
+    ) -> juniper::FieldResult<Command> {
+        let active_mgs_host_fqdn = active_mgs_host_fqdn(&fsname, &context.pg_pool)
+            .await?
+            .ok_or(FieldError::new(
+                "Filesystem not found or MGS is not mounted",
+                Value::null(),
+            ))?;
+
+        let kwargs: HashMap<String, String> =
+            vec![("message".into(), "Unmounting snapshot".into())]
                 .into_iter()
-                .map(|x| Snapshot {
-                    snapshot_name: x.snapshot_name,
-                    filesystem_name: x.filesystem_name,
-                    details: vec![],
-                })
                 .collect();
 
-            Ok(snapshots)
+        let jobs = serde_json::json!([{
+            "class_name": "UnmountSnapshotJob",
+            "args": {
+                "fsname": fsname,
+                "name": name,
+                "fqdn": active_mgs_host_fqdn,
+            }
+        }]);
+        let command_id: i32 = iml_job_scheduler_rpc::call(
+            &context.rabbit_pool.get().await?,
+            "run_jobs",
+            vec![jobs],
+            Some(kwargs),
+        )
+        .map_err(ImlApiError::ImlJobSchedulerRpcError)
+        .await?;
+
+        get_command(&context.pg_pool, command_id)
+            .await
+            .map_err(|e| e.into())
+    }
+}
+
+async fn active_mgs_host_fqdn(
+    fsname: &str,
+    pool: &PgPool,
+) -> Result<Option<String>, iml_postgres::sqlx::Error> {
+    let fsnames = &[fsname.into()][..];
+    let maybe_active_mgs_host_id_row = sqlx::query!(
+        r#"
+            select active_host_id from targets where filesystems @> $1 and name='MGS'
+            "#,
+        fsnames
+    )
+    .fetch_optional(pool)
+    .await?;
```

Comment:
```suggestion
    .await?
    .map(|x| x.active_host_id);
```

To get rid of the row

Reply:
Done